cmd/contour/shutdownmanager.go

```diff
@@ -277,8 +288,8 @@ func registerShutdownManager(cmd *kingpin.CmdClause, log logrus.FieldLogger) (*k
 }
 
 // registerShutdown registers the envoy shutdown sub-command and flags
-func registerShutdown(cmd *kingpin.CmdClause, log logrus.FieldLogger) (*kingpin.CmdClause, *shutdownContext) {
-	ctx := newShutdownContext()
+func registerShutdown(cmd *kingpin.CmdClause, adminPort int, log logrus.FieldLogger) (*kingpin.CmdClause, *shutdownContext) {
+	ctx := newShutdownContext(adminPort)
 	ctx.FieldLogger = log.WithField("context", "shutdown")
 
 	shutdown := cmd.Command("shutdown", "Initiate an shutdown sequence which configures Envoy to begin draining connections.")
```

Comment:
You'll also need to duplicate the `admin-port` flag from `registerBootstrap` here, so that `contour shutdown` will pick up the same flag.

```suggestion
	shutdown := cmd.Command("shutdown", "Initiate an shutdown sequence which configures Envoy to begin draining connections.")
	shutdown.Flag("admin-port", "Envoy admin interface port.").IntVar(&ctx.adminPort)
```